include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -59,6 +59,10 @@ MDSPAN_INLINE_FUNCTION constexpr size_t get_actual_static_padding_value() {
   } else {
     return dynamic_extent;
   }
+  // Missing return statement warning from NVCC
+#ifdef __NVCC__
+  return 0;
```

Comment:
can we do `__builtin_unreachable()` instead?  not sure when support was added

Reply:
```
include/experimental/__p2630_bits/submdspan_extents.hpp-140-  // Even with CUDA_ARCH protection this thing warns about calling host function
include/experimental/__p2630_bits/submdspan_extents.hpp:141:  __builtin_unreachable();
include/experimental/__p2630_bits/submdspan_extents.hpp-142-#endif
--
include/experimental/__p2630_bits/submdspan_mapping.hpp-186-#if defined(__NVCC__) && !defined(__CUDA_ARCH__) && defined(__GNUC__)
include/experimental/__p2630_bits/submdspan_mapping.hpp:187:  __builtin_unreachable();
include/experimental/__p2630_bits/submdspan_mapping.hpp-188-#endif
--
include/experimental/__p2630_bits/submdspan_mapping.hpp-301-#if defined(__NVCC__) && !defined(__CUDA_ARCH__) && defined(__GNUC__)
include/experimental/__p2630_bits/submdspan_mapping.hpp:302:  __builtin_unreachable();
include/experimental/__p2630_bits/submdspan_mapping.hpp-303-#endif

Reply:
Ah: but only in combo with 👍 
```
// Suppress spurious warning with NVCC about no return statement.
// This is a known issue in NVCC and NVC++
// Depending on the CUDA and GCC version we need both the builtin
// and the diagnostic push. I tried really hard to find something shorter
// but no luck ...
#if defined __NVCC__
    #ifdef __NVCC_DIAG_PRAGMA_SUPPORT__
        #pragma nv_diagnostic push
        #pragma nv_diag_suppress = implicit_return_from_non_void_function
    #else
      #ifdef __CUDA_ARCH__
        #pragma diagnostic push
        #pragma diag_suppress implicit_return_from_non_void_function
      #endif
    #endif
#elif defined __NVCOMPILER
    #pragma    diagnostic push
    #pragma    diag_suppress = implicit_return_from_non_void_function
#endif
```
and the corresponding pragma pops

Reply:
Note in the extents case the branches return different types. Here the same type is returned no matter what which means this simpler approach works.